families/supplychain/python/sawtooth_supplychain/rest_api/config.py

```diff
@@ -0,0 +1,179 @@
+
+    LOGGER.info("Loading rest api information from config: %s", filename)
+
+    try:
+        with open(filename) as fd:
+            raw_config = fd.read()
+    except IOError as e:
+        raise RestApiConfigurationError(
+            "Unable to load rest api configuration file: {}".format(str(e)))
+
+    toml_config = toml.loads(raw_config)
+
+    invalid_keys = set(toml_config.keys()).difference(
+        ['bind', 'connect', 'timeout', 'db_cnx'])
+    if invalid_keys:
+        raise RestApiConfigurationError(
+            "Invalid keys in rest api config: {}".format(
+                ", ".join(sorted(list(invalid_keys)))))
+    config = RestApiConfig(
+        bind=toml_config.get("bind", None),
+        connect=toml_config.get('connect', None),
+        timeout=toml_config.get('timeout', None),
+        db_cnx=toml_config.get('db_cnx', None)
+    )
+
+    return config
+
+
+def merge_rest_api_config(configs):
+    """
+    Given a list of PathConfig objects, merges them into a single PathConfig,
+    giving priority in the order of the configs (first has highest priority).
+
+    Args:
+        configs: a list of configs to merge
+    Returns:
+        A single config with values:
+            * bind
+            * connect
+            * timeout
+            * db_cnx
+    """
+    bind = None
+    connect = None
+    timeout = None
+    db_cnx = None
+
+    for config in reversed(configs):
+        if config.bind is not None:
+            bind = config.bind
+        if config.connect is not None:
+            connect = config.connect
+        if config.timeout is not None:
+            timeout = config.timeout
+        if config.db_cnx is not None:
+            db_cnx = config.db_cnx
+
+    return RestApiConfig(
+        bind=bind,
+        connect=connect,
+        timeout=timeout,
+        db_cnx=db_cnx)
+
+
+class RestApiConfig:
+    def __init__(self, bind=None, connect=None, timeout=None, db_cnx=None):
+        self._bind = bind
+        self._connect = connect
+        self._timeout = timeout
+        self._db_cnx = db_cnx
+
+    @property
+    def bind(self):
+        return self._bind
+
+    @property
+    def connect(self):
+        return self._connect
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @property
+    def db_cnx(self):
+        return self._db_cnx
+
+    def __repr__(self):
+        return \
+            "{}(bind={}, connect={}, timeout={}, db_cnx={})".format(
+                self.__class__.__name__,
+                repr(self._bind),
+                repr(self._connect),
+                repr(self._timeout),
+                repr(self._db_cnx))
+
+    def to_dict(self):
+        """Returns a dict of the config values.
+        Args:
+            None
+        Returns:
+            An ordered dict of key/value pairs:
+            * bind
+            * connect
+            * timeout
+            * db_cnx
+        """
+        return collections.OrderedDict([
```

Comment:
Good question. On investigation, this method/function is not being used at all. I removed it altogether.